File: Task3_OwnLanguage/src/Main.java
import gen.GoodLanguageGrammarLexer;
import gen.GoodLanguageGrammarParser;
import org.antlr.v4.runtime.CharStream;
import org.antlr.v4.runtime.CommonTokenStream;

import java.io.BufferedWriter;
import java.io.FileWriter;
import java.io.IOException;
import java.util.Scanner;

import static org.antlr.v4.runtime.CharStreams.fromFileName;

public class Main {
    public static void main(String[] args) throws IOException {
        String inputFileName = new Scanner(System.in).next();
        CharStream cs = fromFileName("tests/in/" + inputFileName);
        GoodLanguageGrammarLexer lexer = new GoodLanguageGrammarLexer(cs);
        CommonTokenStream tokenStream = new CommonTokenStream(lexer);
        GoodLanguageGrammarParser parser = new GoodLanguageGrammarParser(tokenStream);
        MyGoodLanguageVisitor visitor = new MyGoodLanguageVisitor();

        String translatedCode = visitor.visit(parser.program());

        int lastDotIndex = inputFileName.lastIndexOf(".");
        String pureFileName = inputFileName.substring(0, lastDotIndex);
        try (BufferedWriter writer = new BufferedWriter(new FileWriter("tests/out/" + pureFileName + ".c"))) {
            writer.write(translatedCode);
        }
        String generateBinaryCommand = "gcc " + "tests/out/" + pureFileName + ".c " + "-o " +
                "tests/binary/" + pureFileName;
        Runtime.getRuntime().exec(generateBinaryCommand);
    }
}

File: Task3_OwnLanguage/src/Main.kt
import gen.GoodLanguageGrammarLexer
import gen.GoodLanguageGrammarParser
import org.antlr.runtime.CommonTokenStream
import org.antlr.v4.runtime.CharStreams
import java.io.File

//fun main() {
//    val input = File("src/main/resources/input.txt").readText()
//    val lexer = GoodLanguageGrammarLexer(CharStreams.fromString(input))
//    val parser = GoodLanguageGrammarParser(CommonTokenStream(lexer))
//    val program = parser.program()
//    val visitor = MyGoodLanguageVisitor()
//    File("src/main/resources/output.txt").writeText(visitor.visit(program))
//}